ffi/capi/dart/package/lib/lib.dart

```diff
@@ -0,0 +1,11466 @@
+// I believe these two lints produce false positives:
+// ignore_for_file: unnecessary_late
+// ignore_for_file: non_native_function_type_argument_to_pointer
+
+import 'dart:convert';
+import 'dart:ffi' as ffi;
+import 'dart:typed_data';
+import 'package:ffi/ffi.dart' as ffi2;
+
+late final ffi.Pointer<T> Function<T extends ffi.NativeType>(String) _capi;
+void init(String path) => _capi = ffi.DynamicLibrary.open(path).lookup;
+
+late final _callocFree = Finalizer(ffi2.calloc.free);
+
+/// An iterator over code point ranges, produced by `ICU4XCodePointSetData` or
+/// one of the `ICU4XCodePointMapData` types
+class CodePointRangeIterator implements ffi.Finalizable {
```

Comment:
All classes ending with `Iterator` could probably actually implement the iterator interface, by providing a `current` getter and a `moveNext()` method. Is this possible to do using diplomat?

Reply:
Yes this will also need a Diplomat annotation or something that the Dart generator can then use. I don't want to rely solely on the `Iterator` name to make something an iterator.

Reply:
Yeah in ICU4X 2.0 we should have iterator support in diplomat and use it everywhere